### PR TITLE
NAS-114752 / 22.12 / Improve loading experience/performance in ix-tables

### PR DESCRIPTION
**Pull request by denysbutenko**
Changes:

- Use ngrx store for users
- Update unit tests
- Sync changes with snapshots store

TODO:

WebSocket Events doesn't dispatch changes for `user.query`. I need to sync users state with new users and updated
- Dispatch `userChanged()` and `userAdded()` actions from User Form. It requires refactoring User Form to use ix-forms.

### CODE REVIEW

File: src/app/pages/account/users/store/user.effects.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+@Injectable()
+export class UserEffects {
+  loadUsers$ = createEffect(() => this.actions$.pipe(
+    ofType(userPageEntered, builtinUsersToggled),
+    switchMap(() => this.store$.pipe(waitForPreferences)),
+    switchMap((preferences) => {
+      let params: QueryParams<User> = [];
+      if (preferences.hideBuiltinUsers) {
+        // TODO: Fix any
+        params = [[['OR', [['builtin', '=', false], ['username', '=', 'root']]]]] as any;
```
Comment (denysbutenko) [inline src/app/pages/account/users/store/user.effects.ts:29]: I'm sorry about that.
If you know how to fix type checking to make this expression valid please let me know.
image

Comment (undsoft) [reply src/app/pages/account/users/store/user.effects.ts:29]: From standpoint of typings, there are too many braces.
```
[['OR', [['builtin', '=', false], ['username', '=', 'root']]]];
```

Comment (denysbutenko) [reply src/app/pages/account/users/store/user.effects.ts:29]: > From standpoint of typings, there are too many braces.
> 
> ```
> [['OR', [['builtin', '=', false], ['username', '=', 'root']]]];
> ```

Yeah, it passes typescript validation but middleware throwing invalid syntax error in this case

Comment (undsoft) [reply src/app/pages/account/users/store/user.effects.ts:29]: Then there is an issue in typings, either in QueryParams or ApiDirectory. You can try to fix it or forcecast with `as unknown as QueryParams<User>` leaving TODO comment.

**Review by undsoft — APPROVED** [review]
Code looks okay.
From the point of functionality, editing a user does not update the table, but it works in next PR, so I'm approving based on that.

**Pull request merged**